lib/charms/opensearch/v0/helper_charm.py

```diff
@@ -167,16 +167,17 @@ def run_cmd(command: str, args: str = None) -> SimpleNamespace:
         command: can contain arguments
         args: command line arguments
     """
+    command_with_args = command
     if args is not None:
-        command = f"{command} {args}"
+        command_with_args = f"{command} {args}"
 
-    command = " ".join(command.split())
+    command_with_args = " ".join(command_with_args.split())
 
     logger.debug(f"Executing command: {command}")
```

Comment:
@reneradoi we may pass a full string with all arguments to these methods. I'd suggest we do right before:
```suggestion
    command = command.split()[0] if command else None
    logger.debug(f"Executing command: {command}")
```

Reply:
This way, we avoid anyone passing: `command --opt=my_secret` as a string to `run_cmd`.

Reply:
Yes that would be even more strict, while my initial suggestion would leave some room for still leaking sensitive information here. On the other hand, this will remove all other (maybe helpful) args too that were given with the first (the command) parameter, effectively resulting in a log line of just `keytool` (e.g.). Then we could even think about completely removing the logging output here.

Reply:
I like René's approach - I find it to be a good compromise between not losing valuable log details, and leaking secrets. 
We should rather go through our code and see where we might be logging secrets and adapting the logging accordingly

Reply:
That is not only a problem for the present, but towards the future where we may not even involved in this project anymore.

I don't like it. We should instead have an extra option, e.g.:
```
def run_cmd(command: str, args: str = None, log_all_args: bool = False)
```
And a disclaimer in the docstring on the risk of secret leaking if arguments containing them are logged.
So, if you want to troubleshoot something, then you can enable it conscious of what is going to happen.

Reply:
In all applications, preventing secrets leaking is a conscious effort by the developer during programming time.
Our case is no different, the same effort / awareness needs happen, where the developer is responsible for being aware of what they log, and split the args in `run_cmd`.

Reply:
@phvalguima I've adjusted the code to only log the command itself, no args.